pkg/plugin/dumpstate.go

```diff
@@ -0,0 +1,255 @@
+package plugin
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"k8s.io/klog/v2"
+
+	"github.com/neondatabase/autoscaling/pkg/api"
+	"github.com/neondatabase/autoscaling/pkg/util"
+)
+
+type dumpStateConfig struct {
+	Port           uint16 `json:"port"`
+	TimeoutSeconds uint   `json:"timeoutSeconds"`
+}
+
+func (c *dumpStateConfig) validate() (string, error) {
+	if c.Port == 0 {
+		return "port", errors.New("value must be > 0")
+	} else if c.TimeoutSeconds == 0 {
+		return "timeoutSeconds", errors.New("value must be > 0")
+	}
+
+	return "", nil
+}
+
+type stateDump struct {
+	Stopped   bool            `json:"stopped"`
+	BuildInfo util.BuildInfo  `json:"buildInfo"`
+	State     pluginStateDump `json:"state"`
+}
+
+func (p *AutoscaleEnforcer) startDumpStateServer(shutdownCtx context.Context) error {
+	// Manually start the TCP listener so we can minimize errors in the background thread.
+	addr := net.TCPAddr{IP: net.IPv4zero, Port: int(p.state.conf.DumpState.Port)}
+	listener, err := net.ListenTCP("tcp", &addr)
+	if err != nil {
+		return fmt.Errorf("Error binding to %v", addr)
+	}
+
+	go func() {
+		mux := http.NewServeMux()
+		util.AddHandler("dump-state: ", mux, "/", http.MethodGet, "<empty>", func(ctx context.Context, body *struct{}) (*stateDump, int, error) {
```

Comment:
I believe body isn't used, I might not declare the variable at all (use `_`), also I'm pretty sure that all empty structs are actually just a singleton in memory, so the pointer is probably more expensive than using a value,) if that's your intent

Reply:
Yeah, this is due to that quirk of `AddHandler` that I mentioned in the PR description. Changing it is on the todo list, but for now `body` has to be a pointer to something. Naming it makes it a little clearer what's going on.